src/System.Runtime/tests/System/StringTests.cs

```diff
@@ -1293,147 +1293,295 @@ public static void IndexOf_AllSubstrings(string s, string value, int startIndex,
         [Fact]
         public static void IndexOf_TurkishI()
         {
-            string s = "Turkish I \u0131s TROUBL\u0130NG!";
-            Helpers.PerformActionWithCulture(new CultureInfo("tr-TR"), () =>
-            {
-                string value = "\u0130";
-                Assert.Equal(19, s.IndexOf(value));
-                Assert.Equal(19, s.IndexOf(value, StringComparison.CurrentCulture));
-                Assert.Equal(4, s.IndexOf(value, StringComparison.CurrentCultureIgnoreCase));
-                Assert.Equal(19, s.IndexOf(value, StringComparison.Ordinal));
-                Assert.Equal(19, s.IndexOf(value, StringComparison.OrdinalIgnoreCase));
-
-                value = "\u0131";
-                Assert.Equal(10, s.IndexOf(value, StringComparison.CurrentCulture));
-                Assert.Equal(8, s.IndexOf(value, StringComparison.CurrentCultureIgnoreCase));
-                Assert.Equal(10, s.IndexOf(value, StringComparison.Ordinal));
-                Assert.Equal(10, s.IndexOf(value, StringComparison.OrdinalIgnoreCase));
-            });
-            Helpers.PerformActionWithCulture(CultureInfo.InvariantCulture, () =>
+            string str = "Turkish I \u0131s TROUBL\u0130NG!";
+
+            RemoteInvoke((s) =>
             {
-                string value = "\u0130";
-                Assert.Equal(19, s.IndexOf(value));
-                Assert.Equal(19, s.IndexOf(value, StringComparison.CurrentCulture));
-                Assert.Equal(19, s.IndexOf(value, StringComparison.CurrentCultureIgnoreCase));
+                CultureInfo originalCulture = CultureInfo.CurrentCulture;
 
-                value = "\u0131";
-                Assert.Equal(10, s.IndexOf(value, StringComparison.CurrentCulture));
-                Assert.Equal(10, s.IndexOf(value, StringComparison.CurrentCultureIgnoreCase));
-            });
-            Helpers.PerformActionWithCulture(new CultureInfo("en-US"), () =>
+                try
+                {
+                    CultureInfo.CurrentCulture = new CultureInfo("tr-TR");
+
+                    string value = "\u0130";
+                    Assert.Equal(19, s.IndexOf(value));
+                    Assert.Equal(19, s.IndexOf(value, StringComparison.CurrentCulture));
+                    Assert.Equal(4, s.IndexOf(value, StringComparison.CurrentCultureIgnoreCase));
+                    Assert.Equal(19, s.IndexOf(value, StringComparison.Ordinal));
+                    Assert.Equal(19, s.IndexOf(value, StringComparison.OrdinalIgnoreCase));
+
+                    value = "\u0131";
```

Comment:
I believe executing all code of this method inside one RemoteInvoke will just work. also you don't have to reset the current culture back after every part of the test